driver-core/src/main/com/mongodb/internal/operation/AsyncOperationHelper.java

```diff
@@ -61,6 +61,10 @@ interface AsyncCallableWithConnection {
         void call(@Nullable AsyncConnection connection, @Nullable Throwable t);
     }
 
+    interface AsyncCallableConnectionWithCallback<T> {
+        void call(@Nullable AsyncConnection connection, SingleResultCallback<T> callback);
```

Comment:
Does this need to be `Nullable`?  I can't think why, since I think the method wouldn't be called if the resource manager fails to get a connection

Reply:
Done